sparse/src/KokkosSparse_gauss_seidel_handle.hpp

```diff
@@ -127,6 +132,14 @@ class GaussSeidelHandle {
   bool is_numeric_called() const { return this->called_numeric; }
 
   // setters
+  void set_execution_space(const HandleExecSpace exec_space) {
+    static bool is_exec_space_set = false;
+    if (!is_exec_space_set) {
+      this->execution_space = exec_space;
```

Comment:
If the user wants to change the execution space, this can only be done once per handle instantiation.